projects/js-packages/publicize-components/src/hooks/use-media-restrictions/test/index.test.js

```diff
@@ -16,48 +20,102 @@ const DUMMY_CONNECTIONS = [
 	},
 ];
 
-const INVALID_TYPES = [ 'imagejpg', 'image/tgif', '', null ];
+const INVALID_TYPES = [ 'imagejpg', 'image/tgif', 'video/mp5', '', null ];
+const INVALID_LENGTH_VIDEOS = [
+	{ mime: 'video/mp4', fileSize: 1000000, length: 2 }, // Too short video
+	{ mime: 'video/mp4', fileSize: 1000000, length: 20000 }, // Too long video
+];
+const INVALID_SIZED_MEDIA = [
+	{ mime: 'image/jpg', fileSize: 10000000 }, // Too big image
+	{ mime: 'video/mp4', fileSize: 100000000000, length: 20 }, // Too big video
+	{ mime: 'video/mp4', fileSize: 10, length: 20 }, // Too small video
+];
 const VALID_MEDIA = [
-	{ size: 20, type: 'image/jpg' },
-	{ size: 3000000, type: 'image/png' },
+	{ mime: 'image/jpg', fileSize: 20 },
+	{ mime: 'image/png', fileSize: 3000000 },
+	{ mime: 'video/mp4', fileSize: 1000000, length: 20 },
 ];
-
-const setup = connections => renderHook( () => useMediaRestrictions( connections ) );
+const ALLOWED_MEDIA_TYPES_ALL = [ 'image/jpeg', 'image/jpg', 'image/png', 'video/mp4' ];
 
 describe( 'useMediaRestrictions hook', () => {
+	const { result, rerender } = renderHook( connections => useMediaRestrictions( connections ), {
+		initialProps: DUMMY_CONNECTIONS,
+	} );
+
 	test( 'maxImageSize returns the best image size available', () => {
-		const { result } = setup( DUMMY_CONNECTIONS );
+		const defaultMaxImageSize = result.current.maxImageSize;
+		rerender( [ { service_name: 'linkedin' } ] );
+		const linkedinMaxImageSize = result.current.maxImageSize;
+		rerender( DUMMY_CONNECTIONS );
 
-		expect( result.current.maxImageSize ).toBe( 4 );
+		expect( defaultMaxImageSize ).toBe( 4 );
+		expect( linkedinMaxImageSize ).toBe( 20 );
 	} );
 
-	test( 'Too big JPG results in file size error', () => {
-		const { result } = setup( DUMMY_CONNECTIONS );
-
-		const validationError = result.current.getValidationError( 10000000, 'image/jpg' );
+	test( 'Video limits are calculated correctly', () => {
+		const defaultVideoLimits = result.current.videoLimits;
+		rerender( [ { service_name: 'twitter' }, { service_name: 'facebook' } ] );
+		const modifiedVideoLimits = result.current.videoLimits;
+		rerender( DUMMY_CONNECTIONS );
 
-		expect( validationError ).toBe( FILE_SIZE_ERROR );
+		expect( defaultVideoLimits ).toStrictEqual( {
+			maxLength: 140,
+			maxSize: 200,
+			minLength: 3,
+			minSize: 0.075,
+		} );
+		expect( modifiedVideoLimits ).toStrictEqual( {
+			maxLength: 140,
+			maxSize: 512,
+			minLength: 0,
+			minSize: 0,
+		} );
 	} );
 
-	test( 'Invalid file type results in file type error', () => {
-		const { result } = setup( DUMMY_CONNECTIONS );
+	test( 'Returns allowed media types', () => {
+		const allAllowedMediaTypes = result.current.getAllowedMediaTypes();
```

Comment:
I'm not sure why but this testing is failing for me here. I think it might be because the hook isn't returning the function anymore.

Reply:
Yeah true, refactored this.